File: maptiles2/maptiles2.go
package main

import "fmt"

func main() {
	var quadkey string
	fmt.Scanf("%s", &quadkey)
	x, y := 0, 0
	for i := 0; i < len(quadkey); i++ {
		x += x
		y += y
		if quadkey[i] == '1' {
			x++
		} else if quadkey[i] == '2' {
			y++
		} else if quadkey[i] == '3' {
			x++
			y++
		}
	}
	fmt.Printf("%d %d %d\n", len(quadkey), x, y)
}

File: maptiles2/maptiles2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class maptiles2 {

    public static void main(String[] args) throws IOException {
        var in = new BufferedReader(new InputStreamReader(System.in));
        String quadkey = in.readLine();
        int x = 0, y = 0;
        for (int i = 0; i < quadkey.length(); i++) {
            x += x;
            y += y;
            switch (quadkey.charAt(i)) {
            case '1':
                x++;
                break;
            case '2':
                y++;
                break;
            case '3':
                x++;
                y++;
                break;
            }            
        }
        System.out.println(quadkey.length() + " " + x + " " + y);
        in.close();
    }
}
